modules/arrow/src/geoarrow/convert-geojson-feature-to-arrow.ts

```diff
@@ -0,0 +1,319 @@
+import {
+  Float64 as ArrowFloat,
+  FixedSizeList,
+  Field as ArrowField,
+  List as ArrowList,
+  Vector as ArrowVector,
+  makeBuilder
+} from 'apache-arrow';
+import {
+  Point,
+  MultiPoint,
+  LineString,
+  MultiLineString,
+  Polygon,
+  MultiPolygon,
+  Feature
+} from 'geojson';
+
+/**
+ * Arrow Field and geometry vector
+ */
+export type GeoArrowReturnType = {
+  field: ArrowField;
+  geometry: ArrowVector;
+};
+
+/**
+ * Convert GeoJson Features to GeoArrow
+ * @param name
+ * @param features
+ * @returns GeoArrowReturnType
+ */
+export function geojsonFeaturesToArrow(name: string, features: Feature[]): GeoArrowReturnType {
+  // get the geometry type by iterating all features
+  // if one geometry is multi-, then the whole geometry is multi-
+  let geometryType = '';
+  for (let i = 0; i < features.length; i++) {
+    const {geometry} = features[i];
+    if (i === 0) {
+      geometryType = geometry.type;
+    }
+    if (geometry.type.includes('Multi')) {
+      geometryType = geometry.type;
+      break;
+    }
+  }
+  if (geometryType === 'Point') {
+    return geojsonPointToArrow(name, features);
+  } else if (geometryType === 'MultiPoint') {
+    return geojsonMultiPointToArrow(name, features);
+  } else if (geometryType === 'LineString') {
+    return geojsonLineStringToArrow(name, features);
+  } else if (geometryType === 'MultiLineString') {
+    return geojsonMultiLineStringToArrow(name, features);
+  } else if (geometryType === 'Polygon') {
+    return geojsonPolygonToArrow(name, features);
+  } else if (geometryType === 'MultiPolygon') {
+    return geojsonMultiPolygonToArrow(name, features);
+  }
+  throw new Error('Unsupported geometry type');
+}
+
+/**
+ * convert GeoJSON Point to geoarrow
+ * @param name geometry field name
+ * @param points GeoJSON Points
+ * @returns GeoArrowReturnType
+ */
+export function geojsonPointToArrow(name: string, points: Feature[]): GeoArrowReturnType {
+  // get dimension from the first point
+  const dimension = (points[0].geometry as Point).coordinates.length;
+  const pointFieldName = dimension === 2 ? 'xy' : 'xyz';
+
+  // get point type
+  const nullable = false;
+  const coordType = new ArrowFloat();
+  const pointType = new FixedSizeList(
+    dimension,
+    new ArrowField(pointFieldName, coordType, nullable)
+  );
+
+  // create a field
+  const metaData: Map<string, string> = new Map([['ARROW:extension:name', 'geoarrow.point']]);
+  const field = new ArrowField(name, pointType, nullable, metaData);
+
+  // make geoarrow builder
+  const builder = makeBuilder({
+    type: pointType,
+    nullValues: [null]
+  });
+
+  // fill builder with coordinates
+  for (let i = 0; i < points.length; i++) {
+    const coords = (points[i].geometry as Point).coordinates;
+    // @ts-ignore
+    builder.append(coords);
+  }
+
+  // build geometry vector
+  const geometry = builder.finish().toVector();
```

Comment:
If you ever want to handle attributes in the future, ideally each column will have identical chunking. Whereas when you let Arrow JS handle the builders, you don't necessarily have control over the chunking of each column.

Reply:
Good point. I was assuming that we if we build the columns in parallel, we can choose to stop "building" after a certain number of rows and then create a RecordBatch from the "partial" vectors for each column, and then start building new colums/vectors again, rinse and repeat?